controllers/eventsource/resource.go

```diff
@@ -397,6 +398,8 @@ func buildService(args *AdaptorArgs) (*corev1.Service, error) {
 			Selector:  args.Labels,
 		},
 	}
+	svc.ObjectMeta.SetAnnotations(args.EventSource.Spec.Service.Metadata.Annotations)
```

Comment:
nil check.

Reply:
This is an empty dict by default, using kubebuilder comments, I will add a comment in the function denoting that

Reply:
I added an if statement because there is a change that the yaml will be nil explicitly